### ISSUE TITLE
Can't create single key keyboard shortcut

### ISSUE THREAD

**Issue by mpereira** [CDrummond/cantata#945]
Hi,

I recently tried to set `spacebar` to be my Play/Pause keyboard shortcut but noticed that Cantata only allows F1-12 keys as single key shortcuts. Maybe I'm doing something wrong?

Mac OS X (El Capitan) 10.11.6 (15G1004)
Cantata 2.0.1

**Steps to reproduce**
1. Open Cantata
2. On the menu bar: click `Cantata` then `Preferences`
3. Click the `shortcuts` tab
4. Selection an action, click the `Custom` radio box
5. Try to input a single key, e.g. `s`, `spacebar`, `1`
6. Doesn't work

**Expected behavior**
Creating single key keyboard shortcuts works.

Thanks for Cantata by the way, it's the best!

**Comment by CDrummond** [comment]
Please, check for previous reports before creating new ones. https://github.com/CDrummond/cantata/issues/842

**Comment by CDrummond** [comment]
Actually, this seems to be a very easy issue to resolve. Will do for 2.1.x

**Comment by CDrummond** [comment]
Done.

**Comment by mpereira** [comment]
Nice, I'll try to build from master tonight, thanks!

Are you currently accepting donations for Cantata?

**Comment by CDrummond** [comment]
I maintain Cantata purely for 'fun' - so,thanks, but no thanks to the donations. Instead, donate whatever you would have sent me to a local charity.

I've not yet tried the changes on OSX - I develop mainly on Linux, then check on Win/Mac before release. However, the code should be the same.

**Comment by mpereira** [comment]
> I maintain Cantata purely for 'fun' - so,thanks, but no thanks to the donations. Instead, donate whatever you would have sent me to a local charity.

Fair enough. I made a donation to a local charity that helps low income children.

> I've not yet tried the changes on OSX - I develop mainly on Linux, then check on Win/Mac before release. However, the code should be the same.

I'm seeing some qt5 dependency errors when running cmake (and not in the mood to figure them out) so I guess I'll keep an eye out for the next release. Thanks again!

**Comment by mpereira** [comment]
Hi @CDrummond,

I built the latest master on macOS (Sierra) 10.12.5 (16F2073). It works fine (including the single key keyboard shortcut you made possible with https://github.com/CDrummond/cantata/commit/0bacfb783a9bfeebd57db33670eb6b67320a8b36) but there are a few UI glitches:
- icons in the application itself show up as squares
![icons in the application itself show up as squares](https://user-images.githubusercontent.com/112426/27459603-9553b1e0-5785-11e7-87c2-2b0359bc45c4.png)
- the preferences header is transparent and doesn't have submenu icons
![the preferences header is transparent and doesn't have submenu icons](https://user-images.githubusercontent.com/112426/27459607-98787efa-5785-11e7-9f88-3f8714b45b73.png)

I don't know if these are expected since I built against master. Is there a known "stable" SHA?

Software:
```
$ cd ~/git/cantata && git rev-parse HEAD
2404e42f3e43be893ce92874b99bf14db3f42ba5
$ qmake --version
QMake version 3.1
Using Qt version 5.9.0 in /Applications/Qt5.9.0/5.9/clang_64/lib
$ mpg123 --version
mpg123 1.25.0
$ taglib-config --version
1.11.1
```
This is the build output: [output](https://github.com/CDrummond/cantata/files/1096362/build_output.txt)

Let me know if you want me to put this in another issue. Also, I'd be happy to provide MacOS builds from now on since I got the environment set up.
